apps/full-site-editing/full-site-editing-plugin/starter-page-templates/page-template-modal/utils/templates-parser.js

```diff
@@ -0,0 +1,81 @@
+/**
+ * External dependencies
+ */
+/* eslint-disable import/no-extraneous-dependencies */
+import { keyBy, mapValues, get } from 'lodash';
+/* eslint-enable import/no-extraneous-dependencies */
+
+/**
+ * WordPress dependencies
+ */
+/* eslint-disable import/no-extraneous-dependencies */
+import { parse as parseBlocks } from '@wordpress/blocks';
+/* eslint-enable import/no-extraneous-dependencies */
+
+/**
+ * Internal dependencies
+ */
+import replacePlaceholders from './replace-placeholders';
+
+const { templates = [], siteInformation = {} } = window.starterPageTemplatesConfig;
+
+let allTemplatesBlockBySlug = mapValues( keyBy( templates, 'slug' ), ( { content, title } ) => ( {
+	blocks: [],
+	content,
+	count: 0,
+	isEmpty: ! content,
+	isParsing: !! content,
+	title,
+} ) );
+
+// Dispatch a global event once all templates have been parsed.
```

Comment:
Is this comment accurate? When I first saw it I didn't understand what was going on here. Maybe:

> Dispatch a global event to indicate templates are being parsed